bin/pip_find_builddeps.py

```diff
@@ -117,6 +117,26 @@ def generate_file_content(builddeps, is_partial):
     return file_content
 
 
+def _parse_requirements_file(builddeps_file):
+    """Find deps requirements-build.in file."""
+    try:
+        with open(builddeps_file) as f:
+            # ignore line comments or comments added after dependency is declared
+            requirement_re = re.compile(r"^\s*([^\s#]+)")
+            matches = (requirement_re.match(line) for line in f)
+            return set(match.group(1) for match in matches if match)
+    except FileNotFoundError:
+        # it's ok if the file doens't exist.
+        return set()
+
+
+def _sanity_check_args(args):
+    if args.only_write_on_update and not args.output_file:
+        argparse.ArgumentParser.error(
+            "--only-write-on-update requires an output-file (-o/--output-file)."
+        )
```

Comment:
I think this needs to be called on an instance
```shell
$ bin/pip_find_builddeps.py requirements.txt --only-write-on-update   
Traceback (most recent call last):
  File "/home/acmiel/RedHat/Cachito/cachito/bin/pip_find_builddeps.py", line 211, in <module>
    main()
  File "/home/acmiel/RedHat/Cachito/cachito/bin/pip_find_builddeps.py", line 173, in main
    _sanity_check_args(args)
  File "/home/acmiel/RedHat/Cachito/cachito/bin/pip_find_builddeps.py", line 135, in _sanity_check_args
    argparse.ArgumentParser.error(
TypeError: ArgumentParser.error() missing 1 required positional argument: 'message'
```

Reply:
whoops, sorry. I used argparser api incorrectly here